pep-0675.rst

```diff
@@ -913,6 +960,246 @@ is documentation, which is easily ignored and often not seen. With
+valid. Option (a) might surprise users by losing the ``Literal[str]``
+type in innocuous uses, e.g., with ``my_literal.capitalize()``. Option
+(b) would be more user-friendly but would require some more work from
+type checkers.
+
+We decided to favor user-friendliness and go with option (b). However,
+if the Steering Council feels the other way, we are willing to go with
+option (a).
+
+Further, we propose updating the stub for ``str`` in typeshed so that
+the methods are overloads with the ``Literal[str]``-preserving
+versions. This means type checkers do not have hardcode
+``Literal[str]`` behavior for each method. This also lets us easily
+support new methods in the future by updating the typeshed stub.
+
+For example, to preserve literal types for the ``capitalize`` method,
+we would change the stub as below:
+
+::
+
+    # before
+    def capitalize(self) -> str: ...
+
+    # after
+    @overload
+    def capitalize(self: Literal[str]) -> Literal[str]: ...
+    @overload
+    def capitalize(self) -> str: ...
+
+The downside of changing ``str`` stub is that the stub becomes more
+complicated and can make error messages harder to understand. Type
+checkers may need to special-case ``str`` so that error messages are
+understandable for users.
+
+If the Steering Council is opposed to this typeshed stub change, we
+will require type checkers to hardcode these methods.
+
+Below is an exhaustive list of ``str`` methods which, when called as
+indicated with ``Literal[str]``(s) must be treated as returning a
+``Literal[str]``. If this PEP is accepted, we will update these method
+signatures in typeshed:
+
+::
+
+    @overload
+    def __new__(cls, object: Literal[str] = ...) -> Literal[str]: ...
+    @overload
+
+    @overload
+    def capitalize(self: Literal[str]) -> Literal[str]: ...
+    @overload
+    def capitalize(self) -> str: ...
+
+    @overload
+    def casefold(self: Literal[str]) -> Literal[str]: ...
+    @overload
+    def casefold(self) -> str: ...
+
+    @overload
+    def center(self: Literal[str], __width: SupportsIndex, __fillchar: Literal[str] = ...) -> Literal[str]: ...
+    @overload
+    def center(self, __width: SupportsIndex, __fillchar: str = ...) -> str: ...
+
+    if sys.version_info >= (3, 8):
+        @overload
+        def expandtabs(self: Literal[str], tabsize: SupportsIndex = ...) -> Literal[str]: ...
+        @overload
+        def expandtabs(self, tabsize: SupportsIndex = ...) -> str: ...
+
+    else:
+        @overload
+        def expandtabs(self: Literal[str], tabsize: int = ...) -> Literal[str]: ...
+        @overload
+        def expandtabs(self, tabsize: int = ...) -> str: ...
+
+    @overload
+    def format(self: Literal[str], *args: Literal[str], **kwargs: Literal[str]) -> Literal[str]: ...
+    @overload
+    def format(self, *args: str, **kwargs: str) -> str: ...
+
+    @overload
+    def format_map(self: Literal[str], map: Mapping[str, Literal[str]]) -> Literal[str]: ...
```

Comment:
This feels unsafe, since we don't know the exact keys. I'd suggest taking out this one.